tests/aws/services/events/test_archive_and_replay.py

```diff
@@ -0,0 +1,971 @@
+            aws_client.events.describe_archive(ArchiveName=not_existing_archive_name)
+
+        snapshot.add_transformer(
+            [snapshot.transform.regex(not_existing_archive_name, "<archive-name>")]
+        )
+        snapshot.match("describe-archive-unknown-archive-error", error)
+
+    @markers.aws.validated
+    @pytest.mark.skipif(is_old_provider(), reason="not supported by the old provider")
+    def test_list_archive_error_unknown_source_arn(
+        self, region_name, account_id, aws_client, snapshot
+    ):
+        not_existing_event_bus_name = f"doesnotexist-{short_uid()}"
+        non_existing_event_bus_arn = (
+            f"arn:aws:events:{region_name}:{account_id}:event-bus/{not_existing_event_bus_name}"
+        )
+        with pytest.raises(Exception) as error:
+            aws_client.events.list_archives(EventSourceArn=non_existing_event_bus_arn)
+
+        snapshot.add_transformer(
+            [snapshot.transform.regex(not_existing_event_bus_name, "<event-bus-name>")]
+        )
+        snapshot.match("list-archives-unknown-event-bus-error", error)
+
+    @markers.aws.validated
+    @pytest.mark.skip(reason="not possible to test with localstack")
+    def test_update_archive_error_unknown_archive(self, aws_client, snapshot):
+        not_existing_archive_name = f"doesnotexist-{short_uid()}"
+        with pytest.raises(Exception) as error:
+            aws_client.events.update_archive(ArchiveName=not_existing_archive_name)
+
+        snapshot.add_transformer(
+            [snapshot.transform.regex(not_existing_archive_name, "<archive-name>")]
+        )
+        snapshot.match("update-archive-unknown-archive-error", error)
+
+    @markers.aws.validated
+    @pytest.mark.skipif(is_old_provider(), reason="not supported by the old provider")
+    def test_delete_archive_error_unknown_archive(self, aws_client, snapshot):
+        not_existing_archive_name = f"doesnotexist-{short_uid()}"
+        with pytest.raises(Exception) as error:
+            aws_client.events.delete_archive(ArchiveName=not_existing_archive_name)
+
+        snapshot.add_transformer(
+            [snapshot.transform.regex(not_existing_archive_name, "<archive-name>")]
+        )
+        snapshot.match("delete-archive-unknown-archive-error", error)
+
+
+class TestReplay:
+    @markers.aws.validated
+    @pytest.mark.skipif(is_old_provider(), reason="not supported by the old provider")
+    @pytest.mark.parametrize("event_bus_type", ["default", "custom"])
+    @pytest.mark.skip_snapshot_verify(paths=["$..State"])
+    def test_start_list_describe_canceled_replay(
+        self,
+        event_bus_type,
+        region_name,
+        account_id,
+        events_create_event_bus,
+        events_put_rule,
+        create_sqs_events_target,
+        put_event_to_archive,
+        aws_client,
+        snapshot,
+    ):
+        event_start_time = datetime.now(timezone.utc)
+        event_end_time = event_start_time + timedelta(minutes=1)
```

Comment:
this is the timing offset, because executing the lines after will never take one minute. I would keep it as is now since this never caused any flakiness.

Reply:
Although probably valid in this context, I think it's dangerous to make such assumptions and violates a variation of [R07](https://github.com/localstack/localstack/blob/master/docs/testing/README.md#r07). Such assumptions would cause problems if we ever pick up some kind of concrete event simulation or test improvement tooling.